### PR TITLE
Add IPsec Tests

### PR DESCRIPTION
**Pull request by alagoutte**
It is the first step... 

I will add after other tests for IPsec Stats https://github.com/alagoutte/powernsx/tree/Add-Get-NsxIPsecStats

Or when add new feature (for add/set/remove VPN Site)

### CODE REVIEW

File: tests/integration/16.IPsec.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    Context "ipsec" {
+
+        #Group related tests together.
+
+        it "IPsec is Disabled (by default)" {
+
+            #By default IPsec on Edge is disabled (and need to add new "Site" for enable) but no yet PowerNSX cmdlet for this
+            #Check if IPsec is disable
+            (Get-NsxEdge $ipsecedge1name).features.ipsec.enabled | should be false
```
Comment (nmbradford) [inline tests/integration/16.IPsec.Tests.ps1]: What are you testing here?  If you are trying to validate that the env is _read_ to test, then that should not be in a test itself, but in the VeforeEach/All block.  Your tests should concentrate on 'When I issues this command, does the response indicate that what I asked to happen, actually did.  So your test should be something like it "can enable IPSet"  and contain the code to both enable it, and test that it is enabled by examining the response (or the result of a new get)

If this is just a place holder and you arent finished yet, maybe dont submit the PR until you have a set of commits to introduce new functionality and the tests that go with them.

Comment (alagoutte) [reply tests/integration/16.IPsec.Tests.ps1]: at origine, i will enable IPsec but after some test, we need to add IPsec Site for this.. and there is no yet code...

i will plan after to update #354 commit to use this test...
i will prefer to merge #354 before start to work on other stuff...

Comment (nmbradford) [reply tests/integration/16.IPsec.Tests.ps1]: Lets keep it simple.  You need to add tests in #354, that test the functionality being added in #354.  When you have done that, I can merge it.  If you are planning on adding functionality being tested in this PR, then add it and update the tests to test it and let me know when done so I can take a look.

Comment (alagoutte) [reply tests/integration/16.IPsec.Tests.ps1]: I want to add a initial test on this PR and after update this file (and #354) to add specific IPsec Stats tests...
and it is to known if it is good start...

Comment (nmbradford) [reply tests/integration/16.IPsec.Tests.ps1]: Well - refer to my first comment, but yes, the expected approach is to copy the template file as you have done and then populate with suitable tests.  When you have done this, let me know and I will comment.  :)

**Comment by alagoutte** [comment]
Continue on PR #354

**Pull request closed without merging**